src/parse/init.go

```diff
@@ -58,7 +58,25 @@ func newAspParser(state *core.BuildState) *asp.Parser {
 }
 
 func (p *aspParser) ParseFile(state *core.BuildState, pkg *core.Package, filename string) error {
-	return p.asp.ParseFile(pkg, filename)
+	if pkg.Name == "" {
+		return p.asp.ParseFile(pkg, filename, "")
+	}
+
+	subincludes := make([]string, 0, len(state.Config.Parse.PreloadSubincludes))
```

Comment:
can you pull all this logic out into the "constructor"? feels like we shouldn't need to do it every time